src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -324,6 +330,41 @@ private static IEnumerable<VisualStudioInstance> GetInstances(VisualStudioInstan
         }
 
 #if NET46
+        /// <summary>
+        /// Gets CoreXT instances.
+        /// </summary>
+        /// <param name="visualStudioInstances">Optional <see cref="IList{VisualStudioInstance} "/> containing instances.  If none are specified, they are discovered through the Visual Studio Setup API.
+        /// This is only here for unit tests</param>
+        /// <param name="directoryExists">An optional <see cref="Func{String,Boolean} "/> to use when checking directory existence.  Should only be set by unit tests.</param>
+        /// <returns>A <see cref="VisualStudioInstance" /> object representing the Visual Studio instance if one is found, otherwise null.</returns>
+        internal static VisualStudioInstance GetCoreXTInstance(IList<VisualStudioInstance> visualStudioInstances = null, Func<string, bool> directoryExists = null)
+        {
+            var toolset = Environment.GetEnvironmentVariable("MsBuildToolset")?.Trim();
+
+            if (string.IsNullOrWhiteSpace(toolset) || !int.TryParse(toolset, out int toolsetVersion) || toolsetVersion < 150)
+            {
+                // MsBuildToolset is not set or is less than 150
+                return null;
+            }
+
+            var msbuildPath = Environment.GetEnvironmentVariable($"MSBuildToolsPath_{toolset}");
```

Comment:
Does this normally end in a slash? Do we use CoreXT on case-sensitive OSs? Does `Environment.GetEnvironmentVariable` care?